### DISCUSSION TITLE
Chromium inside Appwrite function

### DISCUSSION THREAD

**Discussion by Maxbrt18** [appwrite/appwrite#4888] [Q&A] [answered]
I managed to pass the chromium download with a modification of the environment variables, but when I run my function, I need to access the following binary (/usr/bin/chromium-browser). and I get an error telling me that: no executable was found.

Is it possible to access files in this folder in an Appwrite function? If yes how?

Here is the complete error

```
Error: Tried to find the browser at the configured path (/usr/bin/chromium-browser), but no executable was found.
    at ChromeLauncher.resolveExecutablePath (/usr/code-start/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:98:23)
    at ChromeLauncher.executablePath (/usr/code-start/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:166:25)
    at ChromeLauncher.launch (/usr/code-start/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:70:37)
    at async module.exports (/usr/code-start/src/index.js:5:21)
    at async /usr/local/src/server.js:68:13
```

The problem I think is that Appwrite functions are in a Docker, and in this Docker you can't access host data. Is it possible with mounting to access this data ?

**Comment by stnguyen90** [comment]
> The problem I think is that Appwrite functions are in a Docker, and in this Docker you can't access host data. 

Yes, you are correct that the function runs in an isolated docker container that does not have access to the host. This is function as designed to protect the host and the other containers from the runtime container.

> Is it possible with mounting to access this data ?

You won't be able to mount because the executor spins up the container and you can't really control what gets mounted.

One workaround might be to include the executable as part of the function code that is uploaded to Appwrite. If the executable needs to be at `/usr/bin/chromium-browser` you could move the executable as part of your function code.

**Reply by Maxbrt18** [reply]
Thanks for you answer, does that mean that I should create a folder in my function project directory (for example : bin) in which I store the executable : chromium-browser and just upload it to Appwrite and then copy it for example with this program : 

```
const fs = require('fs');
const path = require('path');

function copyExecutable() {
  const src = path.resolve(__dirname, 'path/to/chromium-browser/executable/in/project/folder');
  const dest = '/usr/bin/chromium-browser';

  fs.copyFile(src, dest, (err) => {
    if (err) {
      console.error(err);
      return;
    }
    console.log(`Successfully copied executable to ${dest}`);
  });
}
```

**Reply by stnguyen90** [reply]
💯 give it a go to see if it works 🤞🏼

**Comment by Maxbrt18** [comment, marked as answer]
After several attempts, I don't think this is possible. The function is 72 MB (keeping the binary compressed) and I have a timeout (after 900s) during the build. I hope that in future updates it will be possible to install manually. Anyway, thanks for your help.
